src/core/public/chrome/ui/header/nav_link.tsx

```diff
@@ -57,7 +57,7 @@ export function createEuiListItem({
         !isModifiedOrPrevented(event)
       ) {
         event.preventDefault();
-        navigateToApp(id);
+        navigateToUrl(url);
```

Comment:
No, we are not parsing this id back anywhere, this is just to prevent deep link id collisions between apps. 
And yes, it is an option to use `navigateToApp` here, but as the id is not parsed anywhere else I thought it was better not to start doing it here if we can use the `url` for the same purpose.